### PR TITLE
Copter: AP_Arming: Added check for EKF origin altitude

### PR DESCRIPTION
**Pull request by arshPratap**
I have added an additional altitude check in the Copter::far_from_EKF_origin as suggested by @rmackay9  for issue #15800.

### CODE REVIEW

File: ArduCopter/commands.cpp [hunk 1]
```diff
@@ -107,7 +107,7 @@ bool Copter::far_from_EKF_origin(const Location& loc)
 {
     // check distance to EKF origin
     Location ekf_origin;
-    if (ahrs.get_origin(ekf_origin) && (ekf_origin.get_distance(loc) > EKF_ORIGIN_MAX_DIST_M)) {
+    if (ahrs.get_origin(ekf_origin) && (ekf_origin.get_distance(loc) > EKF_ORIGIN_MAX_DIST_M) && (abs(ekf_origin.alt - loc.alt) > EKF_ORIGIN_MAX_DIST_M)) {
```
Comment (peterbarker) [inline ArduCopter/commands.cpp:110]: This might need to be a separate limit.

50km is a long way vertically :-)

@rmackay9 's call, 'though

**Review by peterbarker — CHANGES_REQUESTED** [review]
Review message needs some help:

`Copter: AP_Arming: Added check for EKF origin altitude`

File: ArduCopter/commands.cpp (hunk 1 above)
Comment (arshPratap) [reply ArduCopter/commands.cpp:110]: Thank you sir for pointing that out to me..Will a limit of 1Km be an ideal threshold for altitude here ??

Comment (arshPratap) [reply ArduCopter/commands.cpp:110]: I do the necessary changes

Comment (arshPratap) [reply ArduCopter/commands.cpp]: Sir, I have worked on the suggested changes. Please take a look. Thanks!

**Comment by arshPratap** [comment]
Sir, I have worked on the suggested changes. Please take a look. Thanks!

**Review by peterbarker — APPROVED** [review]
Technically good, needs @rmackay9 to be happy before merge.

**Review by khancyr — APPROVED** [review]
Lgtm

**Comment by rmackay9** [comment]
Thanks for this. I think 1km is probably too tight a check.  We are only trying to protect against numerical errors so the original 50km is fine I think.  I think you can just stick with the original definition instead of adding a new one.  thanks!

**Comment by arshPratap** [comment]
Ok, sir, I will rectify that

**Comment by arshPratap** [comment]
Sir, I have added the original 50km value for the altitude check.

File: ArduCopter/commands.cpp (hunk 1 above)
Comment (rmackay9) [inline ArduCopter/commands.cpp:110]: I think this "&&" should be "||" because we want it to know if either the horizontal or vertical distance is too far.

Comment (arshPratap) [reply ArduCopter/commands.cpp]: Got it, Sir

**Comment by arshPratap** [comment]
Sir, I have worked on suggestions. Please take a look . Thanks!

**Comment by rmackay9** [comment]
@arshPratap, Sorry, I keep noticing small things.  So it needs to be "labs()" not "abs()" because the altitude is an int32_t.

**Comment by arshPratap** [comment]
No issues sir, I will change it immediately

**Comment by arshPratap** [comment]
Sir, I have made the changes. Please take a look. Thanks

**Comment by rmackay9** [comment]
Merged, thanks!

**Pull request merged**